### PR TITLE
iodav2 to v3 script

### LINKED ISSUE #58: iodav2 -> v3 script
Create a script to automatically convert iodav2 -> iodav3 based on one-liner from Sherrie. Should run on same yaml file template as s3-copy for consistency if possible. 

- [ ] get one-liner from Sherrie
- [ ] write script
- [ ] test script

### PR DESCRIPTION
**Pull request by jrknezha**
adds a new version of s3-copy as s3-iodav2-v3.py to pull iodav2 files from the bucket, convert to iodav3 via existing scripts on hera, and uploads the new iodav3 file to the bucket

also contains executables needed for iodav3 conversion and a bash script for calling the conversion. 

must be run on hera for the modules for the ioda-upgrade executable.

### CODE REVIEW

File: src/s3-iodav2-v3.py [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+  #----- cycle through the sources
+  si = -1
+  for source in sources.values():
+    si = si + 1
+    # construct source filename
+    fn_template = source["file_template"]["prefix"]+source["file_template"]["root"]+source["file_template"]["suffix"]
+    source_filename = current_cycle.strftime(fn_template)
+    source_dict = {"Bucket" : source["bucket"], "Key" : current_cycle.strftime(source["key"]+fn_template)}
+
+    # check first if destination exists and it is the same as source
```
Comment (frolovsa) [inline src/s3-iodav2-v3.py]: is this logic related to pre-existing destination file still relevant? Seems like if source and destination are the same, that is a problem (they should be different after the conversation).

Comment (frolovsa) [reply src/s3-iodav2-v3.py]: looks like lines 82 through 96 can be removed?

File: src/s3-iodav2-v3.py [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+destionation_file_template = destination["file_template"]["prefix"]+destination["file_template"]["root"]+destination["file_template"]["suffix"]
+source_buckets = []
+for source in sources.values():
+  source_buckets.append( boto3.Session(profile_name=source["credential_profile"]).resource('s3').Bucket(source["bucket"]))
+print('...Done')
+
+#----- open log files
+print('Opening loggers')
+dtg = datetime.now().strftime('%Y%m%d%H%M%S')
+logger_present = open(config_dict["logging"]["present"].format(dtg=dtg),'wt')
```
Comment (frolovsa) [inline src/s3-iodav2-v3.py]: maybe this log file is not needed if the comment below is valid?

**Review by frolovsa — CHANGES_REQUESTED** [review]
Thank you @jrknezha for making it work. 

I requested the change because i dont think the logic about existing destination file applies here. By design, we expect the source and destination files be different.

File: src/s3-iodav2-v3.py (hunk 1 above)
Comment (jrknezha) [reply src/s3-iodav2-v3.py]: yes, excellent point, they aren't relevant to this script. I have removed them

File: src/s3-iodav2-v3.py (hunk 2 above)
Comment (jrknezha) [reply src/s3-iodav2-v3.py]: agreed and removed

**Review by frolovsa — APPROVED** [review]
thumbs up!

**Pull request merged**